### ISSUE TITLE
The variable input size for fit

### ISSUE THREAD

**Issue by Peydon** [fchollet/keras#7450]
hello, i recently use a CNN+LSTM model to do some NLP tasks ，but the input size of sequences is variable.I try the LSTM model with masking layer and it works well. In fact ,my model doesn't  require a fixed length of input and the docs also say Conv1D can solve the variable input. However ,the fit need the fixed shape of trainx and trainy. One solution is separate the different length of sequences and make up
a list of data of fixed shape. And you also can train one sequence  each time but the result is worse than LSTM which in a paper it should be better. My question is since the model can accept different length,why
the fit function still check the input shape?

**Comment by KARABAER** [comment]
well,now i get a good solution.I use the fit_generator() function to define my own fit batch.I just yield a batch size of the same length sequence with the shape (bacth_size,len,dim)。it works well, and the LSTM model can use this skill  without masking layer.
